client/state/data-layer/wpcom/comments/index.js

```diff
@@ -58,6 +117,16 @@ export const addComments = ( { dispatch }, { siteId, postId }, next, { comments,
 	}
 };
 
+export const writePostCommentSuccess = ( { dispatch, getState }, { siteId, postId, parentCommentId, placeholderId }, next, comment ) => {
+	// remove placeholder from state
+	dispatch( { type: COMMENTS_REMOVE, siteId, postId, commentId: placeholderId } );
+	// add new comment to state with updated values from server
+	dispatch( { type: COMMENTS_RECEIVE, siteId, postId, comments: [ comment ], skipSort: !! parentCommentId } );
+	// increment comments count
+	const commentCount = getPostTotalCommentsCount( getState(), siteId, postId );
+	dispatch( { type: COMMENTS_COUNT_RECEIVE, siteId, postId, totalCommentsCount: commentCount + 1 } );
```

Comment:
So to summarize, dispatching `ADD_COMMENT` in `writePostComment` and `UPDATE_COMMENT` on success and `COMMENTS_REMOVE` on failure?

That does sound like a cleaner separation there. If we add an `UPDATE_COMMENT` action, do you think we still need a `COMMENTS_REMOVE`?

```jsx
{
     type: ADD_COMMENT, 
     siteId, 
     postId, 
     comment, 
     skipSort: !! parentCommentId 
}
```

```jsx
{
     type: UPDATE_COMMENT, 
     siteId, 
     postId,
     commentId,
     comment,
}
```